akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala

```diff
@@ -166,13 +176,17 @@ private[akka] case class ActorFlowMaterializerImpl(override val settings: ActorF
     session.materialize().asInstanceOf[Mat]
   }
 
-  lazy val executionContext: ExecutionContextExecutor = dispatchers.lookup(settings.dispatcher match {
+  override lazy val executionContext: ExecutionContextExecutor = dispatchers.lookup(settings.dispatcher match {
     case Deploy.NoDispatcherGiven ⇒ Dispatchers.DefaultDispatcherId
     case other                    ⇒ other
   })
 
-  private[akka] def actorOf(props: Props, name: String): ActorRef =
-    actorOf(props, name, settings.dispatcher)
+  override def actorOf(context: MaterializationContext, props: Props): ActorRef = {
+    val dispatcher =
+      if (props.dispatcher == Dispatchers.DefaultDispatcherId) context.effectiveSettings.dispatcher
```

Comment:
props.deploy.dispatcher == NoDispatcherGiven

If a Props is using the default dispatcher explicitly then we should not override it